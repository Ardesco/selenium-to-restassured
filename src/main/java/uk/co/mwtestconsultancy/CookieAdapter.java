package uk.co.mwtestconsultancy;

import com.jayway.restassured.response.Cookie;

import java.util.Calendar;
import java.util.TimeZone;
import java.util.concurrent.TimeUnit;

import static java.lang.Math.toIntExact;

public class CookieAdapter {

    private ExpiryType expiryType;

    public CookieAdapter() {
        TimeZone.setDefault(TimeZone.getTimeZone("GMT"));
    }

    public CookieAdapter(ExpiryType expiryType){
        TimeZone.setDefault(TimeZone.getTimeZone("GMT"));
        this.expiryType = expiryType;
    }

    public org.openqa.selenium.Cookie convertToSelenium(Cookie cookie) {
        org.openqa.selenium.Cookie.Builder newCookie = new org.openqa.selenium.Cookie.Builder(cookie.getName(), cookie.getValue());

        if(cookie.getComment() != null){
            System.out.println("NOTE: The cookie taken from RestAssured has a comment attribute of '" + cookie.getComment() + "'. Selenium doesn't support the addition of comments so this has been discarded");
        }

        if(cookie.getDomain() != null){
            newCookie.domain(cookie.getDomain());
        }

        if(cookie.getExpiryDate() != null){
            newCookie.expiresOn(cookie.getExpiryDate());
        }

        if(cookie.isHttpOnly()){
            newCookie.isHttpOnly(cookie.isHttpOnly());
        }

        if(cookie.getMaxAge() != -1){
            Calendar currentDate = Calendar.getInstance();
            currentDate.setTimeZone(TimeZone.getTimeZone("GMT"));
            currentDate.add(Calendar.SECOND, cookie.getMaxAge());

            newCookie.expiresOn(currentDate.getTime());
        }

        if(cookie.getPath() != null){
            newCookie.path(cookie.getPath());
        }

        if(cookie.isSecured()){
            newCookie.isSecure(cookie.isSecured());
        }

        if(cookie.getVersion() != -1){
            System.out.println("NOTE: The cookie taken from RestAssured has a version attribute of '" + cookie.getVersion() + "'. Selenium doesn't support the addition of version so this has been discarded");
        }

        return newCookie.build();
    }

    public Cookie convertToRestAssured(org.openqa.selenium.Cookie cookie) {
        Cookie.Builder newCookie = new Cookie.Builder(cookie.getName(), cookie.getValue());

        if(cookie.getDomain() != null){
            newCookie.setDomain(cookie.getDomain());
        }

        if(cookie.getExpiry() != null){
            if(expiryType.equals(ExpiryType.EXPIRY)){
                newCookie.setExpiryDate(cookie.getExpiry());
            } else if(expiryType.equals(ExpiryType.MAXAGE)){
                Calendar futureDate = Calendar.getInstance();
                futureDate.setTime(cookie.getExpiry());
                Calendar currentDate = Calendar.getInstance();

                long end = TimeUnit.MILLISECONDS.toSeconds(futureDate.getTimeInMillis());
                long start = TimeUnit.MILLISECONDS.toSeconds(currentDate.getTimeInMillis());
                int result = toIntExact(end - start);

                newCookie.setMaxAge(result);
            }
        }

        if(cookie.isHttpOnly()){
            newCookie.setHttpOnly(cookie.isHttpOnly());
        }

        if(cookie.isSecure()){
            newCookie.setSecured(cookie.isSecure());
        }

        if(cookie.getPath() != null){
            newCookie.setPath(cookie.getPath());
        }

        return newCookie.build();
    }
}
