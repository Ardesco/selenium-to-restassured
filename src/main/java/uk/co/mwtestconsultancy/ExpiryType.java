package uk.co.mwtestconsultancy;

public enum ExpiryType {
    EXPIRY,
    MAXAGE
}
