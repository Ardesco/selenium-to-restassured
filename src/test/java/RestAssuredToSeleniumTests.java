import io.restassured.http.Cookie;
import io.restassured.response.Response;
import org.approvaltests.Approvals;
import org.junit.Test;
import uk.co.mwtestconsultancy.CookieAdapter;

import java.util.Calendar;
import java.util.TimeZone;

import static io.restassured.RestAssured.given;
import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.MatcherAssert.assertThat;

public class RestAssuredToSeleniumTests {

    @Test
    public void convertBasicCookieTest(){
        Cookie restAssuredCookie = new Cookie.Builder("Cookie name", "Cookie value").build();

        CookieAdapter cookieAdapter = new CookieAdapter();
        org.openqa.selenium.Cookie adaptedCookie = cookieAdapter.convertToSelenium(restAssuredCookie);

        Approvals.verify(adaptedCookie.toString());
    }

    @Test
    public void convertComplexCookieTest(){
        Calendar currentDate = Calendar.getInstance();
        currentDate.set(2018, 01, 01);
        currentDate.set(Calendar.HOUR, 0);
        currentDate.set(Calendar.MINUTE, 0);
        currentDate.set(Calendar.SECOND, 0);
        currentDate.set(Calendar.HOUR_OF_DAY, 0);

        Cookie restAssuredCookie = new Cookie.Builder("Cookie name", "Cookie value")
                                        .setComment("Cookie comment")
                                        .setDomain("www.mwtestconsultancy.co.uk")
                                        .setExpiryDate(currentDate.getTime())
                                        .setHttpOnly(true)
                                        .setPath("/test")
                                        .setSecured(false)
                                        .setVersion(1)
                                        .build();

        CookieAdapter cookieAdapter = new CookieAdapter();
        org.openqa.selenium.Cookie adaptedCookie = cookieAdapter.convertToSelenium(restAssuredCookie);

        Approvals.verify(adaptedCookie);
    }

    @Test
    public void convertMaxAgeCookieTest(){
        Calendar currentDate = Calendar.getInstance();
        currentDate.setTimeZone(TimeZone.getTimeZone("GMT"));
        currentDate.add(Calendar.SECOND, 86400);


        Cookie restAssuredCookie = new Cookie.Builder("Cookie name", "Cookie value")
                                            .setMaxAge(86400)
                                            .build();

        CookieAdapter cookieAdapter = new CookieAdapter();
        org.openqa.selenium.Cookie adaptedCookie = cookieAdapter.convertToSelenium(restAssuredCookie);

        assertThat(adaptedCookie.getExpiry().toString(), is(currentDate.getTime().toString()));
    }

    @Test
    public void convertLiveCookieTest(){
        Response response = given()
                                .get("http://the-internet.herokuapp.com/");

        CookieAdapter cookieAdapter = new CookieAdapter();
        org.openqa.selenium.Cookie convertedCookie = cookieAdapter.convertToSelenium(response.getDetailedCookie("rack.session"));

        Approvals.verify(convertedCookie.getClass());
    }

}
