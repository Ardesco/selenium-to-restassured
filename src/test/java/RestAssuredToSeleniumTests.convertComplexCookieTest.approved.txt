Cookie name=Cookie value; expires=Thu, 01 Feb 2018 12:00:00 GMT; path=/test; domain=www.mwtestconsultancy.co.uk
